opencensus/stats/aggregation.py

```diff
@@ -14,125 +14,79 @@
-      SUM (int): The aggregation type of the view is 'sum'.
-      COUNT (int): The aggregation type of the view is 'count'.
-      DISTRIBUTION (int): The aggregation type of the view is 'distribution'.
-      LASTVALUE (int): The aggregation type of the view is 'lastvalue'.
-    """
-    NONE = 0
-    SUM = 1
-    COUNT = 2
-    DISTRIBUTION = 3
-    LASTVALUE = 4
-
-
-class BaseAggregation(object):
-    """Aggregation describes how the data collected is aggregated by type of
-    aggregation and buckets
-
-    :type buckets: list(:class: '~opencensus.stats.bucket_boundaries.
-                                BucketBoundaries')
-    :param buckets: list of endpoints if the aggregation represents a
-                    distribution
-
-    :type aggregation_type: :class:`~opencensus.stats.aggregation.Type`
-    :param aggregation_type: represents the type of this aggregation
-
-    """
-    def __init__(self, buckets=None, aggregation_type=Type.NONE):
-        self._aggregation_type = aggregation_type
-        self._buckets = buckets or []
-
-    @property
-    def aggregation_type(self):
-        """The aggregation type of the current aggregation"""
-        return self._aggregation_type
-
-    @property
-    def buckets(self):
-        """The buckets of the current aggregation"""
-        return self._buckets
-
-
-class SumAggregation(BaseAggregation):
-    """Sum Aggregation escribes that data collected and aggregated with this
+class SumAggregation(object):
+    """Sum Aggregation describes that data collected and aggregated with this
     method will be summed
 
     :type sum: int or float
-    :param sum: the sum of the data collected and aggregated
-
-
-    :type aggregation_type: :class:`~opencensus.stats.aggregation.Type`
-    :param aggregation_type: represents the type of this aggregation
+    :param sum: the initial sum to be used in the aggregation
 
     """
-    def __init__(self, sum=None, aggregation_type=Type.SUM):
-        super(SumAggregation, self).__init__(aggregation_type=aggregation_type)
-        self._sum = aggregation_data.SumAggregationDataFloat(
-            sum_data=float(sum or 0))
-        self.aggregation_data = self._sum
-
-    @property
-    def sum(self):
-        """The sum of the current aggregation"""
-        return self._sum
+    def __init__(self, sum=None):
+        self._initial_sum = sum or 0
+
+    def new_aggregation_data(self, measure):
+        """Get a new AggregationData for this aggregation."""
+        value_type = MetricDescriptorType.to_type_class(
+            self.get_metric_type(measure))
+        return aggregation_data.SumAggregationData(
+            value_type=value_type, sum_data=self._initial_sum)
+
+    @staticmethod
+    def get_metric_type(measure):
+        """Get the MetricDescriptorType for the metric produced by this
+        aggregation and measure.
+        """
+        if isinstance(measure, measure_module.MeasureInt):
```

Comment:
I don't have strong opinion here, it seems a more "Pythonic" style is:
```python
        if isinstance(measure, measure_module.MeasureInt):
            return MetricDescriptorType.CUMULATIVE_INT64
        if isinstance(measure, measure_module.MeasureFloat):
            return MetricDescriptorType.CUMULATIVE_DOUBLE
        raise ValueError
```
Up to you.

Reply:
Good catch, surprised the linter didn't yell about this (or maybe it was a warning I didn't see :grin:)